go/types/opcache.go

```diff
@@ -8,15 +8,33 @@ import (
 	"encoding/binary"
 	"io/ioutil"
 	"os"
+	"sync"
+	"sync/atomic"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/hash"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
 )
 
-func newOpCache(vrw ValueReadWriter) *opCache {
+var (
+	collectionId = uint32(0)
+	ldb          *opCacheDb
```

Comment:
Before, each opCache had a defined lifetime: it was bound to a given mutator instance. Now, if any instance is alive, then this whole thing lives. It also means you can't inject some other opCache implementation for testing.

Also, if your 10GB number is accurate, it means that each open levelDB instance adds 33MB of unique memory overhead? That seems like a lot. I know that there are some caches that each instance uses. Perhaps we could reduce those in size for these opcaches and see if that changes things. There's `DefaultBlockCacheCapacity`, which defaults to 8MB. We think that's for reads, but we can't actually decipher the code and comments. There's also `DefaultWriteBuffer`, which starts at 4MB. Other stuff: https://godoc.org/github.com/syndtr/goleveldb/leveldb/opt